share/availability/full/reconstruction_test.go

```diff
@@ -79,79 +118,119 @@ func TestShareAvailable_OneFullNode(t *testing.T) {
 // source, full node must be able to finish the availability process started in
 // the beginning.
 func TestShareAvailable_ConnectedFullNodes(t *testing.T) {
-	// NOTE: Numbers are taken from the original 'Fraud and Data Availability Proofs' paper
-	light.DefaultSampleAmount = 20 // s
-	const (
-		origSquareSize = 16 // k
-		lightNodes     = 60 // c
-	)
-
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
-	defer cancel()
-
-	net := availability_test.NewTestDAGNet(ctx, t)
-	source, root := RandNode(net, origSquareSize)
-
-	// create two full nodes and ensure they are disconnected
-	full1 := Node(net)
-	full2 := Node(net)
-
-	// pre-connect fulls
-	net.Connect(full1.ID(), full2.ID())
-	// ensure fulls and source are not connected
-	// so that fulls take data from light nodes only
-	net.Disconnect(full1.ID(), source.ID())
-	net.Disconnect(full2.ID(), source.ID())
-
-	// start reconstruction for fulls
-	errg, errCtx := errgroup.WithContext(ctx)
-	errg.Go(func() error {
-		return full1.SharesAvailable(errCtx, root)
-	})
-	errg.Go(func() error {
-		return full2.SharesAvailable(errCtx, root)
-	})
-
-	// create light nodes and start sampling for them immediately
-	lights1, lights2 := make(
-		[]*availability_test.TestNode, lightNodes/2),
-		make([]*availability_test.TestNode, lightNodes/2)
-	for i := 0; i < len(lights1); i++ {
-		lights1[i] = light.Node(net)
-		go func(i int) {
-			err := lights1[i].SharesAvailable(ctx, root)
-			if err != nil {
-				t.Log("light1 errors:", err)
+	tc := []struct {
+		name            string
+		origSquareSize  int  // k
+		lightNodes      int  // c
+		sampleAmount    uint // s
+		recoverability  Recoverability
+		expectedFailure bool
+	}{
+		{
+			name:            "fully recoverable",
+			origSquareSize:  16,
+			lightNodes:      24,
+			sampleAmount:    20,
+			recoverability:  FullyRecoverable,
+			expectedFailure: false,
+		},
+		{
+			name:            "barely recoverable",
+			origSquareSize:  16,
+			lightNodes:      69,
```

Comment:
If you divide `int` 69 by 2 you will get 34, so total light nodes will be 68. So I assume lower value here should make this test flaky and the value needs to be 70, or need to use uneven devision for light nodes groups.

Reply:
Oh true, I set it to 70 in another case for this reason but forgot to apply everywhere